framework/elements/BaseElement.js

```diff
@@ -0,0 +1,169 @@
+import Logger from "../logConfiguration/Logger.js";
+
+
+class BaseElement {
+    constructor(name, selector) {
+        this.name = name;
+        this.selector = selector;
+    }
+
+    async getElement() {
+        Logger.trace(`Geting ${this.name} element`)
+        return browser.$(this.selector);
+    }
+
+    async getElements(selector) {
+        Logger.trace(`Geting ${this.name} elements`)
+        return browser.$$(selector);
+    }
+
+    async getValue() {
+        Logger.debug(`Getting the value of ${this.name}`)
+        const element = await this.getElement(this.selector);
+        const value = await element.getValue();
+        Logger.debug(`The value of ${element} is ${value}`)
+        return value;
+    }
+
+    async getAttribute(attributeName) {
+        Logger.debug(`Getting the value of attribute ${attributeName}`)
+        const element = await this.getElement(this.selector);
+        const attributeValue = await element.getAttribute(attributeName);
+        Logger.debug(`The value of attribute ${attributeName} of element ${element} is ${attributeValue}`)
+        return attributeValue;
+    }
+
+    async moveTo() {
+        Logger.debug(`Moving element to ${this.name}`)
+        const element = await this.getElement(this.selector);
+        element.moveTo();
+    }
+
+    async isDisplayed() {
+        Logger.debug(`Checking if ${this.name} is diplayed`)
+        const element = await this.getElement();
+        let isDisplayed;
+        try {
+            isDisplayed = await element.isDisplayed();
+        } catch (error) {
+            isDisplayed = false;
+        }
+        Logger.debug(`The result of isDisplayed for ${this.name} is ${isDisplayed}`)
+        return isDisplayed;
+    }
+
+    async isExisting() {
+        Logger.debug(`Checking if ${this.name} exists`)
+        const element = await this.getElement(this.selector);
+        let isExisting;
+        try {
+            isExisting = await element.isExisting();
+        } catch (error) {
+            isExisting = false;
+        }
+        Logger.debug(`The result of isExisting for ${this.name} is ${isExisting}`)
+        return isExisting;
+    }
+
+    async isEnabled() {
+        Logger.debug(`Checking if ${this.name} is enabled`)
+        const element = await this.getElement(this.selector);
+        let isEnabled;
+        try {
+            isEnabled = await element.isEnabled();
+        } catch (error) {
+            isEnabled = false;
+        }
+        Logger.debug(`The result of isEnabled for ${this.name} is ${isEnabled}`)
+        return isEnabled;
+    }
+
+    async isSelected() {
```

Comment:
Elena, I highly recommend you to include waiting for some condition inside of such methods. It will make your work much easier